platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java

```diff
@@ -16,1037 +16,751 @@
+ * members are forking, the theorem still guarantees at least one famous witness is known by R+3.
  *
- * It is another theorem that the d12 and d2 algorithm have more than two thirds of the population creating
- * unique famous witnesses (judges) in each round. It is a theorem that d1 does, too, for the algorithmn described in
- * 2016, and is conjectured to be true for the 2019 version, too.
+ * <p>It is another theorem that the d12 and d2 algorithm have more than two thirds of the
+ * population creating unique famous witnesses (judges) in each round. It is a theorem that d1 does,
+ * too, for the algorithmn described in 2016, and is conjectured to be true for the 2019 version,
+ * too.
  *
- * Another new theorem used here:
+ * <p>Another new theorem used here:
  *
- * Theorem: If a new witness X is added to round R, but at least one already exists in round R+2, then X
- * will not be famous (so there is no need to hold the elections).
+ * <p>Theorem: If a new witness X is added to round R, but at least one already exists in round R+2,
+ * then X will not be famous (so there is no need to hold the elections).
  *
- * Proof: If an event X currently exists in round R+2, then when the new event Y is added to round R, it
- * won't be an ancestor of X, nor of the witnesses that X strongly sees. Therefore, X will collect unanimous
- * votes of NO for the fame of Y, so X will decide that Y is not famous. Therefore, once a round R+2 event
- * is added to the hashgraph, the set of possible unique famous witnesses for round R is fixed, and the
- * unique famous witnesses will end up being a subset of it.
+ * <p>Proof: If an event X currently exists in round R+2, then when the new event Y is added to
+ * round R, it won't be an ancestor of X, nor of the witnesses that X strongly sees. Therefore, X
+ * will collect unanimous votes of NO for the fame of Y, so X will decide that Y is not famous.
+ * Therefore, once a round R+2 event is added to the hashgraph, the set of possible unique famous
+ * witnesses for round R is fixed, and the unique famous witnesses will end up being a subset of it.
  *
- * NOTE: for concision, all of the above talks about things like "2/3 of the members" or "2/3 of the witnesses". In
- * every case, it should be interpreted to actually mean "members whose weight adds up to more than 2/3 of the total
- * weight", and "witnesses created by members whose weight is more than 2/3 of the total".
- **/
-public class ConsensusImpl implements Consensus {
+ * <p>NOTE: for concision, all of the above talks about things like "2/3 of the members" or "2/3 of
+ * the witnesses". In every case, it should be interpreted to actually mean "members whose stake
+ * adds up to more than 2/3 of the total stake", and "witnesses created by members whose stake is
+ * more than 2/3 of the total".
+ */
+public class ConsensusImpl extends ThreadSafeConsensusInfo implements Consensus {
```

Comment:
I don't understand the decision to extend `ThreadSafeConsensusInfo` instead of using an instance of it. IMO, it's not a good reason to extend something just so that the data can be accessed using fewer characters. I was ok with this approach with `EventImpl` because we plan on doing a complete refactor at some point, but the same is not true here.

Reply:
The current implementation (in develop) has a mix of thread-safe and non thread-safe data in it. I think we all agree this needs to change. 
Ideally, this would be in some separate instance that other threads can access it. But making this change now would have a lot of knock-on effects. So I decided to make it a superclass for now to limit the scope of this PR. We can discuss this further.